metaflow/metadata/heartbeat.py

```diff
@@ -43,16 +48,29 @@ def ping(self):
 
                 time.sleep(frequency_secs)
                 retry_counter = 0
-            except Exception:
+            except HeartBeatException as e:
                 retry_counter = retry_counter + 1
                 time.sleep(4**retry_counter)
 
     def heartbeat(self):
         if self.hb_url is not None:
             response = \
                 requests.post(url=self.hb_url, data="{}", headers=self.headers)
-            return response.json().get('wait_time_in_seconds')
+            # Unfortunately, response.json() returns a string that we need
+            # to cast to json; however when the request encounters an error
+            # the return type is a json blob :/
+            if response.status_code == 200:
+                return json.loads(response.json()).get('wait_time_in_seconds')
+            else:
+                raise HeartBeatException('HeartBeat request (%s) failed'
+                                         ' (code %s): %s' % 
+                                            (self.hb_url, response.status_code,
+                                            response.text))
+        return None
 
     def shutdown(self):
-        # attempts sending on last heartbeat
-        self.heartbeat()
+        # attempts sending one last heartbeat
+        try:
```

Comment:
comment from @ferras - https://github.com/Netflix/metaflow/pull/516/files#r628725650

Reply:
main point of that comment was to say that you can centralize the error/exception handling here:
https://github.com/Netflix/metaflow/blob/master/metaflow/sidecar_worker.py#L23
so that it applies to all sidecars if the desire is to suppress messages from the user